pkg/cmd/describe/describe_rollout.go

```diff
@@ -0,0 +1,533 @@
+	case *appsv1.StatefulSet:
+		workloadInfo.Replicas.Desired = *o.Spec.Replicas
+		workloadInfo.Replicas.Current = o.Status.Replicas
+		workloadInfo.Replicas.Updated = o.Status.UpdatedReplicas
+		workloadInfo.Replicas.Ready = o.Status.ReadyReplicas
+		workloadInfo.Replicas.Available = o.Status.AvailableReplicas
+	case *kruiseappsv1alpha1.DaemonSet:
+		workloadInfo.Replicas.Desired = o.Spec.BurstReplicas.IntVal
+		workloadInfo.Replicas.Current = o.Status.CurrentNumberScheduled
+		workloadInfo.Replicas.Updated = o.Status.UpdatedNumberScheduled
+		workloadInfo.Replicas.Ready = o.Status.NumberReady
+		workloadInfo.Replicas.Available = o.Status.NumberAvailable
+	case *kruiseappsv1beta1.StatefulSet:
+		workloadInfo.Replicas.Desired = *o.Spec.Replicas
+		workloadInfo.Replicas.Current = o.Status.Replicas
+		workloadInfo.Replicas.Updated = o.Status.UpdatedReplicas
+		workloadInfo.Replicas.Ready = o.Status.ReadyReplicas
+		workloadInfo.Replicas.Available = o.Status.AvailableReplicas
+	case *kruiseappsv1alpha1.CloneSet:
+		workloadInfo.Replicas.Desired = *o.Spec.Replicas
+		workloadInfo.Replicas.Current = o.Status.Replicas
+		workloadInfo.Replicas.Updated = o.Status.UpdatedReplicas
+		workloadInfo.Replicas.Ready = o.Status.ReadyReplicas
+		workloadInfo.Replicas.Available = o.Status.AvailableReplicas
+	default:
+		return nil, fmt.Errorf("unsupported workload kind %T", obj)
+	}
+
+	workloadInfo.CurrentRevision = rollout.Status.CanaryStatus.StableRevision
+	workloadInfo.UpdateRevision = rollout.Status.CanaryStatus.CanaryRevision
+
+	var labelSelectorParam string
+	switch obj.(type) {
+	case *appsv1.Deployment:
+		labelSelectorParam = "pod-template-hash"
+	default:
+		labelSelectorParam = "controller-revision-hash"
+	}
+
+	// Fetch pods
+	r2 := o.Builder().
+		WithScheme(internalapi.GetScheme(), scheme.Scheme.PrioritizedVersionsAllGroups()...).
+		NamespaceParam(o.Namespace).DefaultNamespace().
+		ResourceTypes("pods").
+		LabelSelectorParam(fmt.Sprintf("%s=%s", labelSelectorParam, rollout.Status.CanaryStatus.PodTemplateHash)).
+		Latest().
+		Flatten().
+		Do()
+
+	if err := r2.Err(); err != nil {
+		return nil, err
+	}
+
+	err = r2.Visit(func(info *resource.Info, err error) error {
+		if err != nil {
+			return err
+		}
+
+		pod, ok := info.Object.(*corev1.Pod)
+		if !ok {
+			return fmt.Errorf("expected *corev1.Pod, got %T", info.Object)
+		}
+
+		podInfo := struct {
+			Name     string
+			BatchID  string
+			Status   string
+			Ready    string
+			Age      string
+			Restarts string
+			Revision string
+		}{
+			Name:     pod.Name,
+			BatchID:  pod.Labels["rollouts.kruise.io/rollout-batch-id"],
+			Status:   string(pod.Status.Phase),
+			Age:      duration.HumanDuration(time.Since(pod.CreationTimestamp.Time)),
+			Restarts: fmt.Sprintf("%v (%v ago)", strconv.Itoa(int(pod.Status.ContainerStatuses[0].RestartCount)), duration.HumanDuration(time.Since(pod.Status.ContainerStatuses[0].LastTerminationState.Terminated.FinishedAt.Time))),
```

Comment:
this line will panic : 
1. pod.status.containerstatus may be empty for a newly created pod
2. pod.status.containerstatus[0].LastTerminationState.Terminated may also be null